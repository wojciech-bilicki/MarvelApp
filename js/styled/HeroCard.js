import styled from "styled-components";

const HeroCard = styled.div`
  width: 24px;
  height: 28px;
  background-color: blue;
`;

export default HeroCard;
