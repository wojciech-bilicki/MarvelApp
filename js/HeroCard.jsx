// @flow
import React from 'react';
import styled from 'styled-components';
import type { Hero } from './types';

const HeroCardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  background-color: #fff;
  align-items: center;
  justify-content: center;
  margin: 16px;
  box-shadow: 3px 2px 6px #666666;
  max-width: 200px;

  img {
    width: 150px;
    height: 200px;
  }

  h4 {
    font-size: 18px;
    font-weight: bold;
  }

  h5,
  h4 {
    padding: 8px 0;
    word-wrap: break-word;
  }

  button {
    color: #ffffff;
    font-size: 20px;
    background: #3f34d9;
    padding: 10px 20px 10px 20px;
    text-decoration: none;
    border: none;
    margin-bottom: 8px;
  }
`;

const HeroCard = ({ hero }: { hero: Hero }) =>
  <HeroCardWrapper>
    <h4>
      {hero.name}
    </h4>
    <img src={`../assets/portraits/${hero.image}`} alt="Hero Portrait" />
    <h5>
      {hero.true_name}
    </h5>
    <button>Add to my heroes</button>
  </HeroCardWrapper>;

export default HeroCard;
