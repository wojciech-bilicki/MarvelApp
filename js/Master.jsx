// @flow
import React from "react";
import styled from "styled-components";
import HeroCard from "./HeroCard";
import type {Hero} from './types';

const CardListWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

type Props = {
  searchTerm: string,
  heroes: Array<Hero>
}

const Master = (props: Props) => (
  <div>
    <CardListWrapper>
      {props.heroes
        .filter(hero => {
          if (!props.searchTerm || props.searchTerm === "") {
            return true;
          }
          return (
            `${hero.name} ${hero.true_name}`
              .toUpperCase()
              .indexOf(props.searchTerm.toUpperCase()) >= 0
          );
        })
        .map(hero => <HeroCard hero={hero} key={hero.id} />)}
    </CardListWrapper>
  </div>
);

export default Master;
