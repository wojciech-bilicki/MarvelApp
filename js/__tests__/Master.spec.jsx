import React from "react";
import { shallow } from "enzyme";

import Master from "../Master";
import HeroCard from "../HeroCard";
import data from "../../data.json";

test("Master renders properly", () => {
  const master = shallow(<Master heroes={data.heroes} />);
  expect(master).toMatchSnapshot();
});

xtest("Master render HeroCard amount based on props", () => {
  const master = shallow(<Master heroes={data.heroes} />);
  expect(master.find(HeroCard).length).toEqual(data.heroes.length);
});
