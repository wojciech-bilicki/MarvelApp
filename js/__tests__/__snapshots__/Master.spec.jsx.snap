// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Master renders properly 1`] = `
<styled.div>
  <h2>
    Dude, wft I am loading!
  </h2>
  <MyHeroes
    favouriteHeroes={Array []}
    onDelete={[Function]}
  />
</styled.div>
`;
